cli/js/timers_test.ts

```diff
@@ -134,7 +134,7 @@ test(async function intervalCancelSuccess(): Promise<void> {
   assertEquals(count, 0);
 });
 
-test(async function intervalOrdering(): Promise<void> {
+test(async function timeoutOrdering(): Promise<void> {
```

Comment:
This test currently fails 😢 

The problem is that we create 11 timeouts here, and interestingly first created timeout fires last, all other timeouts are called in proper order

Reply:
Okay, after some investigation this is what's happening here:
1. all timeouts are scheduled with 1 ms delay
2. [`Delay`](https://docs.rs/tokio/0.1.22/tokio/timer/struct.Delay.html) has resolution of 1 ms
3. We don't order pending futures
4. Random timeout is fired after 1 ms because of the resolution.

So problem to solve is to guarantee ordering of timeouts. The simplest one would be to use  `Delay` per RBTree node instead of `Delay` per timeout.

Reply:
Or increase the timeout time in this test. I'd be fine with that. I don't think it's of much consequence.

Reply:
Actually @piscisaureus suggested that browsers and Node ensure timeout ordering, so it's better to keep it. I'll try with delay per RBTree implementation to see how it works out